src/modules/household/components/ManageHousehold.tsx

```diff
@@ -123,6 +129,9 @@ const ManageHousehold = ({
     [onSuccess, refetchHousehold]
   );
 
+  // could happen if user entered bad URL for /projects/:id/add-household/:householdId
+  if (householdNotFound) return <NotFound />;
```

Comment:
When the household is not found, the page flickers for a second before resolving on Page not found.

Untested, but would it be possible to avoid the flicker by moving what's now line 151 up here to before line 133? `if (householdId && !household && loading) return <Loading />`?

Reply:
thanks good point, updating